### PR TITLE
6 get api article comments

### PR DESCRIPTION
**Pull request by Jay-R-Mac**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -127,7 +128,107 @@ describe("GET /api/articles", () => {
...
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_id: expect.any(Number)
+          })
+        }))
+
+      });
+  });
+  it("responds with an array which contains the articles comments in ascending order", () => {
```
Comment (Jodiech) [inline __tests__/app.test.js]: Make sure your description matches what you are asserting in your test. Is it ascending or descending? 😄

File: __tests__/app.test.js [hunk 2]
```diff
@@ -127,7 +128,107 @@ describe("GET /api/articles", () => {
...
+      .then(({ text }) => {
+        expect(JSON.parse(text)).toEqual({ message: "Article Not Found" });
       });
   });
+  it("responds with a 400 status and a message when an invalid id is passed", () => {
+    return request(app)
+      .get("/api/articles/hello/comments")
+      .expect(400)
+      .then(({ res }) => expect(res.statusMessage).toBe("Bad Request"));
+  });
```
Comment (Jodiech) [inline __tests__/app.test.js:179]: What happens if there is a valid ID but no comments? Please add in a test to check for this, you should expect that this will respond with an empty array.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -127,7 +128,107 @@ describe("GET /api/articles", () => {
...
+        expect(body.length).toBe(2)
+        body.forEach((comment => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            body: expect.any(String),
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_id: expect.any(Number)
+          })
```
Comment (Jodiech) [inline __tests__/app.test.js:152]: You don't need to include the article_id here, this can be removed 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -127,7 +128,107 @@ describe("GET /api/articles", () => {
...
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({body}) => {
+        expect(body.body).toBe("This is my test article comment")
+      });
+      
+  });
+  it("responds with a 404 and a message when article is not found", () => {
```
Comment (Jodiech) [inline __tests__/app.test.js]: This description doesn't match the behaviour. I'd change this to: responds with a 404 when passed with a non existent ID

File: __tests__/app.test.js [hunk 5]
```diff
@@ -127,7 +128,107 @@ describe("GET /api/articles", () => {
...
+      .send(newComment)
+      .expect(400)
+      .then(({ res }) => expect(res.statusMessage).toBe("Bad Request"));
+  });
+  it("responds with a 400 status and a message when an no username is passed", () => {
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(noUserComment)
+      .expect(400)
+      .then(({ res }) => expect(res.text).toBe("You must provide a username"));
```
Comment (Jodiech) [inline __tests__/app.test.js]: These two 400 tests can be combined into one. I'd change the description to: **'should respond with a status 400 when missing required fields, e.g. no username or body properties passed'**

File: __tests__/app.test.js [hunk 6]
```diff
@@ -127,7 +128,107 @@ describe("GET /api/articles", () => {
...
+  });
+  it("responds with a 404 and a message when article is not found", () => {
+    return request(app)
+      .post("/api/articles/200000/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ text }) => {
+        expect(JSON.parse(text)).toEqual({ message: "Article Not Found" });
+      });
+  });
```
Comment (Jodiech) [inline __tests__/app.test.js:217]: Please add another test for a 201 - ignoring unnecessary properties.

File: model/model.js [hunk 7]
```diff
@@ -42,4 +42,42 @@ function getArticles() {
...
+    })
+
+}
+
+function postComment(article, newComment) {
+  const { username, body } = newComment
+  const author = username
+  const article_id = article
+  return db.query("SELECT * FROM articles WHERE article_id = $1;", [article_id])
+    .then(({ rows }) => {
```
Comment (Jodiech) [inline model/model.js:65]: This isn't necessary as you will be posting a comment, please delete this first query!

**Review by Jodiech — CHANGES_REQUESTED** [review]
Hey 👋 nice work on this ticket. I'd like you to make a few changes and rebsumit. You need to add in a test for a valid ID but no comments for ticket 6. This is going to need some functionality added in too, so let me know if you have any questions.

File: __tests__/app.test.js [hunk 8]
```diff
@@ -176,39 +175,43 @@ describe("GET /api/articles/:article_id/comments", () => {
       .expect(400)
       .then(({ res }) => expect(res.statusMessage).toBe("Bad Request"));
   });
+  it("responds with a 404 and a message when article is not found", () => {
+    return request(app)
+      .get("/api/articles/4/comments")
+      .expect(404)
+      .then(({ text }) => {
+        expect(JSON.parse(text)).toEqual({ message: "Article Not Found or No Comments Exist" });
```
Comment (NRMSMITH) [inline __tests__/app.test.js]: Ideally, this should display an empty array - your frontend will handle the 'error message'. As far as we're concerned, this is really a 200 - because it's not an error to not have comments

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
A change to a test :D

File: __tests__/app.test.js [hunk 9]
```diff
@@ -127,7 +128,107 @@ describe("GET /api/articles", () => {
...
+  });
+  it("responds with a 400 status and a message when an invalid id is passed", () => {
+    return request(app)
+      .get("/api/articles/hello/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("Bad Request");
       });
   });
+  it("responds with a 404 and a message when article is not found", () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Please change status code in the description to 200.

File: __tests__/app.test.js [hunk 10]
```diff
@@ -127,7 +128,107 @@ describe("GET /api/articles", () => {
...
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ text }) => {
+        expect(text).toEqual("No Comments Yet");
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Please expect an empty array here.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Jordan, please see my comments, amend your code, then it's good to go.
